File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int M = sc.nextInt();
        int N = sc.nextInt();
        int A[] = new int[N+1];
        for(int i = 2; i < N+1; i++){
            A[i] = i;
        }
        for(int j = 2; j<= Math.sqrt(N); j++){
            if(A[j] == 0 ) continue;
            for(int i = j+j ; i <= N; i=i+j){
                A[i] = 0;
            }
        }
        for(int i = M; i<=N; i++){
            if(A[i] == 0) continue;
            System.out.println(i);
        }
    }
}

File: Python/백준/Silver/1929. 소수 구하기/소수 구하기.py
import math

m, n= map(int, input().split())
A=[True]*(n+1)
A[0] = A[1] = False

for i in range(2, int(math.sqrt(n)) + 1):
    if A[i]:
        for j in range( i*i, n+1, i):
            A[j] = False

for i in range(m, n + 1):
    if A[i]:
        print(i)